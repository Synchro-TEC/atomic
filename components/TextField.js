import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const TextField = ({ label, onChange, children, ...otherProps }) => {
  const baseClass = `sv-tag ${theme}`;
  const rootClass = cx(baseClass, className);

  return (
    <label>
      <span data-info="(8 caracters)" data-error="Cep invÃ¡lido">
        {label}
      </span>
      <input type="text" className="is--invalid" onChange={onChange} />
    </label>
  );
};

TextField.defaultProps = {
  theme: 'default',
};

TextField.propTypes = {
  label: PropTypes.string,
  onClose: PropTypes.func,
  theme: PropTypes.oneOf(['default', 'primary', 'danger', 'info', 'warning']),
};

export default TextField;
