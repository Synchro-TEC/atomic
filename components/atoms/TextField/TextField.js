import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const TextField = ({ label, disabled, type, onChange, hintInfo, errorMessage, ...otherProps }) => {
  let labelOpts = {};

  if (hintInfo) {
    labelOpts['data-info'] = hintInfo;
  }

  if (errorMessage) {
    labelOpts['data-error'] = errorMessage;
  }

  const rootClass = cx({ 'is--invalid': errorMessage });

  let inputOptions = {
    type: type,
    onChange: onChange,
    className: rootClass,
  };

  if (disabled) {
    inputOptions.disabled = disabled;
  }

  const inputProps = Object.assign(inputOptions, otherProps);

  return (
    <label>
      <span {...labelOpts}>{label}</span>
      <input {...inputProps} />
    </label>
  );
};

TextField.defaultProps = {
  type: 'text',
  disabled: false,
};

TextField.propTypes = {
  errorMessage: PropTypes.string,
  disabled: PropTypes.bool,
  hintInfo: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  type: PropTypes.oneOf(['text', 'email', 'password']),
};

export default TextField;
