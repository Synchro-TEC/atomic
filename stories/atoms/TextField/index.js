import React from 'react';
import { action } from '@storybook/addon-actions';
import readme from './README.md';
import Row from '../../../components/Row';
import Col from '../../../components/Col';
import TextField from '../../../components/TextField';
import { text, boolean } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../utils/optionalSelect';

const options = {
  text: 'text',
  email: 'email',
  password: 'password',
};

const component = () => (
  <form className="sv-form">
    <Row withGutter={true}>
      <Col>
        <TextField label="Name" errorMessage={text('Name error:', '')} onChange={action('on change name')} />
      </Col>
      <Col>
        <TextField label="E-mail" hintInfo={text('E-mail hint:', '')} onChange={action('on change email')} />
      </Col>
      <Col>
        <TextField
          label="Phone"
          hintInfo={text('Phone hint:', '')}
          errorMessage={text('Phone error', '')}
          onChange={action('on change phone')}
        />
      </Col>
    </Row>
    <Row withGutter={true}>
      <Col>
        <TextField
          label="Dynamic Field"
          value="test"
          onChange={action('on change Dynamic Field')}
          type={optionalSelect('Type', options, '')}
          disabled={boolean('Disabled', false)}
        />
      </Col>
    </Row>
  </form>
);

export default [readme, component];
