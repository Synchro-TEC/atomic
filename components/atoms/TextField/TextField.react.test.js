import React from 'react';
import TextField from './TextField';

describe('TextField Component', () => {
  test('Render the TextField', () => {
    const textField = shallow(<TextField />);
    expect(textField).toMatchSnapshot();
  });

  test('Fire change on TextField', () => {
    const onChange = jest.fn();
    const textFieldWrapper = shallow(<TextField onChange={onChange} />);
    textFieldWrapper.find('input').simulate('change', () => {});
    expect(onChange.mock.calls.length).toEqual(1);
  });
});
