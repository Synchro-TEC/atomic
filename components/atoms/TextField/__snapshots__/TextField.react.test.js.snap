// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component Render the TextField 1`] = `
<label>
  <span />
  <input
    className=""
    type="text"
  />
</label>
`;
